incubator/binding-pgsql-kafka/src/main/java/io/aklivity/zilla/runtime/binding/pgsql/kafka/internal/schema/PgsqlKafkaAvroSchemaTemplate.java

```diff
@@ -91,13 +90,18 @@ protected void applyAlterations(
     {
         for (AlterExpression alterExpr : alterExpressions)
         {
-            if (Objects.requireNonNull(alterExpr.operation()) == Operation.ADD)
+            if (alterExpr.operation() == Operation.ADD)
             {
                 ObjectNode newField = mapper.createObjectNode();
                 newField.put("name", alterExpr.columnName());
                 newField.set("type", mapper.valueToTree(mapSqlTypeToAvroType(alterExpr.columnType())));
                 fields.add(newField);
             }
+            else
+            {
+                throw new UnsupportedOperationException(
+                    String.format("Unsupported alter operation: %s", alterExpr.operation()));
```

Comment:
This is using exceptions for flow control, which we typically avoid because it is heavy.